invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/fields/Identifiers/PIDField/components/OptionalDOIoptions.js

```diff
@@ -42,13 +42,18 @@ class OptionalDOIoptionsCmp extends Component {
     ].includes(record.status);
 
     const hasDoi = doi !== "";
-    const isUnManagedDisabled = alreadyPublished
-      ? !optionalDOItransitions.allowed_providers.includes("external")
-      : hasDoi && isManagedSelected;
-
-    const isNoNeedDisabled = alreadyPublished
-      ? !optionalDOItransitions.allowed_providers.includes("not_needed")
-      : hasDoi && isManagedSelected;
+    const doiManaged = hasDoi && isManagedSelected;
+
+    function isDisabled(provider) {
+      return (
+        doiManaged ||
```

Comment:
disable the options "Yes, I have" and "No, I need one" in the UI if there is locally managed DOI reserved. We do that so that the user is forced to unreserve first the managed DOI.